include/jemalloc/internal/jemalloc_internal_inlines_c.h

```diff
@@ -19,6 +19,21 @@
 #define TCACHE_IND_AUTOMATIC ((unsigned)-2)
 #define ARENA_IND_AUTOMATIC ((unsigned)-1)
 
+/*
+ * Under normal circumstances, whether or not to use a slab
+ * to satisfy an allocation depends solely on the allocation's
+ * effective size. However, this is *not* the case when an allocation
+ * is sampled for profiling, in which case you *must not* use a slab
+ * regardless of the effective size. Thus `should_use_slab` is called
+ * on the common path, but there exist `*_explicit_slab` variants of
+ * several functions below for handling the aforementioned case of
+ * sampled allocations.
+ */
+JEMALLOC_ALWAYS_INLINE bool
+should_use_slab(size_t usize) {
```

Comment:
We can't always do this it turns out. Or at least, in `iralloct` we should be calling:
```
bool slab = sz_can_use_slab(sz_sa2u(size, alignment))
```

Because `iralloct_realign` applies `sz_sa2u` to `usize` before passing it down, which causes the ` assert(!slab || sz_can_use_slab(usize)); /* slab && large is illegal */` to fail.

Reply:
I see. In that case, let's stick to `usize`, and make sure that we always pass in an usize into it.